packages/expo-updates/android/src/main/java/expo/modules/updates/launcher/DatabaseLauncher.kt

```diff
@@ -145,8 +145,28 @@ class DatabaseLauncher(
     return selectionPolicy.selectUpdateToLaunch(filteredLaunchableUpdates, manifestFilters)
   }
 
-  internal fun ensureAssetExists(asset: AssetEntity, database: UpdatesDatabase, context: Context): File? {
-    val assetFile = File(updatesDirectory, asset.relativePath)
+  private fun embeddedAssetFileMap(context: Context): MutableMap<AssetEntity, String> {
+    val embeddedManifest = EmbeddedManifest.get(context, this.configuration)
+    val embeddedAssets: List<AssetEntity> = embeddedManifest?.assetEntityList ?: listOf()
+    return mutableMapOf<AssetEntity, String>().apply {
+      for (asset in embeddedAssets) {
+        if (asset.isLaunchAsset) {
+          continue
+        }
+        val filename = asset.relativePath
+        if (filename != null) {
+          val file = when (asset.embeddedAssetFilename != null) {
+            true -> File(asset.embeddedAssetFilename)
+            false -> File(updatesDirectory, asset.relativePath)
+          }
```

Comment:
```suggestion
          val embeddedAssetFilename = asset.embeddedAssetFilename
          val file = if (embeddedAssetFilename != null) {
            File(embeddedAssetFilename)
          } else {
            File(updatesDirectory, asset.relativePath)
          }
```

(more common in kotlin to use if/else as an expression rather than when with just boolean)